### PR TITLE
APE tool: Trend plots, some minor fixes

### PR DESCRIPTION
**Pull request by mteroerd**
#### PR description:

This PR is the summary of some changes that were made to the APE estimation tool during the UL2017 and UL2018 campaigns. 

It includes:
- a tool for trend plots (hence, the data/ folder and additions in test/plottingTools)
- a way to change marker styles and colors for result plots
- a way to add the number of hits in a sector to result plots
- fix to a small typo in test/batch/startSkim.py
- fix to the wrong ideal alignment being loaded in test/cfgTemplate/apeEstimator_cfg.py
- additional functionality for test/autoSubmitter:
  * Basic support for submitting to condor "CAF queue".
  * Protection against cases where the condor scheduler is unavailable when submitting jobs which used to cause crashes. Now, in this case the submission is retried after some waiting time until it works.
  * Support for running multiple threads to handle measurements in case of large campaigns. Before, it happened that condor jobs (that do the track refits) of  measurements finished while another measurement was being handled, so that there was some delay before these measurements were handled, which piled up to longer delay when doing many measurements. 
   * Some minor changes like putting some functions into helper.py and introducing new shortcuts for convenience.

#### PR Validation:

The validation happened while using the tool. Since no changes were made that could affect the APE results themselves, everything was transparent.

### CODE REVIEW

**Comment by pohsun** [comment]
please test

**Comment by mteroerd** [comment]
I forgot to mention that this will probably also need a backport to the release on which UL2016ReReco will be done.

**Comment by pohsun** [comment]
+1
@mteroerd Please submit a PR for backport and relate to this one.

File: Alignment/APEEstimation/test/plottingTools/trendPlotter.py [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+
+def whichYear(run):
+    thisYear = -1
+    years = runs.keys()
+    years.sort()
+    for year in years:
+        if min(runs[year]) <= run:
+            thisYear = year
+    return thisYear
+    print "Run %d not in range of any year"%(run)
```
Comment (fabiocos) [inline Alignment/APEEstimation/test/plottingTools/trendPlotter.py]: @mteroerd this will fail python3 compilation

Comment (mteroerd) [reply Alignment/APEEstimation/test/plottingTools/trendPlotter.py]: Thanks for the comment, fixed.

**Comment by fabiocos** [comment]
please test

**Comment by pohsun** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Pull request merged**